helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -326,6 +327,21 @@ public T get(String path, Stat stat, int options) {
     return data;
   }
 
+  public <K> K get(String path, Stat stat, int options, GenericSerializer<K> serializer) {
+    //TODO
+    return null;
+  }
+
+  public <K> boolean create(String path, Stat stat, int options, GenericSerializer<K> serializer) {
+    //TODO
+    return false;
+  }
+
+  public <K> boolean set(String path, Stat stat, int options, GenericSerializer<K> serializer) {
+    //TODO
+    return false;
+  }
```

Comment:
Since both PathBasedZkSerializer and ZkSerializer just use Java's generic Object and byte array, we do not need to introduce another generic type K. So it should look like the following:

`public Object get(String path, Stat stat, int options, ZkSerializer (or PathBasedZkSerializer) customSerializer);`

Also, let us split the PR so that we have separate PRs for each public method if you think that's easier?